### PR TITLE
use SmallRye Metrics fixes #968

### LINKED ISSUE #968: SmallRye Metrics integration

### PR DESCRIPTION
**Pull request by michalszynkiewicz**
- [X] Have you followed the guidelines in our [Contributing](http://wildfly-swarm.io/community/contributing/) document?
- [ ] [v2] Have you created a [JIRA](https://issues.jboss.org/browse/THORN) and used it in the commit message?
- [X] [v4] Have you created a [GitHub Issue](https://github.com/thorntail/thorntail/issues) and used it in the commit message?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/thorntail/thorntail/pulls) for the same issue?
- [X] Have you built the project locally prior to submission with `mvn clean install`?

-----

### CODE REVIEW

File: core/metrics/src/main/resources/io/smallrye/metrics/mapping.yml [hunk 1]
```diff
@@ -0,0 +1,110 @@
+base:
```
Comment (kenfinnigan) [inline core/metrics/src/main/resources/io/smallrye/metrics/mapping.yml]: What's the mapping file for, given it wasn't present before?

Comment (michalszynkiewicz) [reply core/metrics/src/main/resources/io/smallrye/metrics/mapping.yml]: It's defining the server specific metrics. V2 used mapping.yml file for it and v4 used properties files. I pulled out the class that parses the yaml config into [SmallRye](https://github.com/smallrye/smallrye-metrics/pull/1/files#diff-c04dd7f3444f13aa6ceeeed1140a63aaR45) and made v4 use it.

**Comment by bobmcwhirter** [comment]
If we could do properties we could avoid the extra dependency on snakeyaml. That’s why v4 moved away from yaml.

**Comment by kenfinnigan** [comment]
Was just about to mention that, beat me to it.

I think SmallRye should also not automatically do YAML, but if YAML is present then do it. If that makes sense

**Comment by michalszynkiewicz** [comment]
okay, I'll try to switch SmallRye to properties.

**Comment by michalszynkiewicz** [comment]
@bobmcwhirter @kenfinnigan could we use a single property file, like https://gist.github.com/michalszynkiewicz/844b9da1f67fb57696ca391ac306e835 for the mapping? It's easier to get hold of it than file per metric

**Comment by bobmcwhirter** [comment]
Sure I just wanted to avoid yet another dependency. This is a-okay.

**Comment by michalszynkiewicz** [comment]
restest this please

**Review by kenfinnigan — APPROVED** [review]

**Pull request merged**